doc/read-the-docs-site/tutorials/english-auction-example.rst

```diff
@@ -0,0 +1,344 @@
+
+Note that ``AuctionParams`` is an argument of neither the untyped validator nor the final UPLC program. As said before, it contains contract properties that don't change, so it is simply built into the validator.
+
+Since the PlutusTx compiler compiles ``a`` into ``CompiledCode a``, we first use ``auctionUntypedValidator`` to obain an untyped validator.
+It takes ``AuctionParams``, and returns an untyped validator.
+We then define the ``auctionValidatorScript`` function, which takes ``AuctionParams`` and returns the compiled UPLC program.
+
+Next, to create the Plutus validator script for a particular auction, we call ``auctionValidatorScript`` with the appropriate ``AuctionParams``.
+We will then be able to launch the auction on-chain by submitting a transaction that locks the asset being auctioned at the script address with ``AuctionDatum Nothing`` as the datum.
+
+.. note::
+   It is worth noting that we must call ``PlutusTx.compile`` on the entire ``auctionUntypedValidator``, rather than applying it to ``params`` before compiling, as in ``$$(PlutusTx.compile [||auctionUntypedValidator params||])``.
+   The latter won't work, because everything being compiled (inside ``[||...||]``) must be known at compile time, but ``params`` is not: it is a formal parameter.
+   Instead, we compile the entire ``auctionUntypedValidator`` into UPLC, then use ``liftCode`` to lift ``params`` into a UPLC term, and apply the compiled ``auctionUntypedValidator`` to it at the UPLC level.
+   To do so, we need the ``Lift`` instance for ``AuctionParams``, derived via ``PlutusTx.makeLift``.
+
+Libraries for Writing Plutus Tx Scripts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This auction example shows a relatively low-level way of writing scripts using Plutus Tx.
+In practice, you may consider using a higher-level library that abstracts away some of the details.
+For example, `plutus-apps <https://github.com/input-output-hk/plutus-apps>`_ provides a state machine library and a constraint library for writing Plutus Tx.
+Using these libraries, writing a validator in Plutus Tx becomes a matter of defining state transactions and the corresponding constraints, e.g., the condition ``refundsPreviousHighestBid`` can simply be written as ``Constraints.mustPayToPubKey bidder (singleton adaSymbol adaToken amt)``.
+
+Off-chain code
+-----------------------
+
+Since the main purpose of this tutorial is to introduce Plutus Tx and Plutus Core, we only walked through the on-chain code, responsible for validating transactions (in the sense of determining whether a transaction is allowed to spend an EUTXO).
+
+In addition to the on-chain code, one typically needs the accompanying off-chain code and services to perform tasks like building transactions, submitting transactions, deploying smart contracts, querying for available EUTXOs on the chain, etc.
```

Comment:
I would also mention in a paragraph `cardano-api` which also provides (in a more low-level way) the capability to create, balancing and submit transactions with a local running node (with even some querying functions like getting the EUTXOs of a particular address).

Reply:
Thanks for this suggestion. I'd like to get a better handle on how cardano-api is used to do these sorts of things. It would be great to be able to at least broadly describe a valid process of using cardano-api to create, balance and submit a transaction. Or perhaps point to another source that may already exist that describes it.